projects/control-service/projects/pipelines_control_service/src/integration-test/java/com/vmware/taurus/datajobs/it/DataJobDeployTemporaryCredsIT.java

```diff
@@ -0,0 +1,174 @@
+
+    DataJobVersion testDataJobVersion =
+        new ObjectMapper()
+            .readValue(jobUploadResult.getResponse().getContentAsString(), DataJobVersion.class);
+    Assertions.assertNotNull(testDataJobVersion);
+
+    String testJobVersionSha = testDataJobVersion.getVersionSha();
+    Assertions.assertFalse(StringUtils.isBlank(testJobVersionSha));
+
+    // Setup
+    String dataJobDeploymentRequestBody = getDataJobDeploymentRequestBody(testJobVersionSha);
+
+    // Execute build and deploy job
+    mockMvc
+        .perform(
+            post(String.format(
+                    "/data-jobs/for-team/%s/jobs/%s/deployments", TEST_TEAM_NAME, TEST_JOB_NAME))
+                .with(user("user"))
+                .content(dataJobDeploymentRequestBody)
+                .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(status().isAccepted());
+
+    String jobDeploymentName = JobImageDeployer.getCronJobName(TEST_JOB_NAME);
+
+    // Verify job deployment created
+    Optional<JobDeploymentStatus> cronJobOptional =
+        dataJobsKubernetesService.readCronJob(jobDeploymentName);
+    Assertions.assertTrue(cronJobOptional.isPresent());
+    JobDeploymentStatus cronJob = cronJobOptional.get();
+    Assertions.assertEquals(testJobVersionSha, cronJob.getGitCommitSha());
+    Assertions.assertEquals(DataJobMode.RELEASE.toString(), cronJob.getMode());
+    Assertions.assertEquals(true, cronJob.getEnabled());
+    Assertions.assertTrue(cronJob.getImageName().endsWith(testJobVersionSha));
+    Assertions.assertEquals("user", cronJob.getLastDeployedBy());
+
+    MvcResult result =
+        mockMvc
+            .perform(
+                get(String.format(
+                        "/data-jobs/for-team/%s/jobs/%s/deployments/%s",
+                        TEST_TEAM_NAME, TEST_JOB_NAME, DEPLOYMENT_ID))
+                    .with(user("user"))
+                    .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isOk())
+            .andReturn();
+
+    // Verify response
+    DataJobDeploymentStatus jobDeployment =
+        mapper.readValue(result.getResponse().getContentAsString(), DataJobDeploymentStatus.class);
+    Assertions.assertEquals(testJobVersionSha, jobDeployment.getJobVersion());
+    Assertions.assertEquals(true, jobDeployment.getEnabled());
+    Assertions.assertEquals(DataJobMode.RELEASE, jobDeployment.getMode());
+    Assertions.assertEquals(true, jobDeployment.getEnabled());
+    // by default the version is the same as the tag specified by datajobs.vdk.image
+    // for integration test this is registry.hub.docker.com/versatiledatakit/quickstart-vdk:release
+    Assertions.assertEquals("release", jobDeployment.getVdkVersion());
+    Assertions.assertEquals("user", jobDeployment.getLastDeployedBy());
+    // just check some valid date is returned. It would be too error-prone/brittle to verify exact
+    // time.
+    DateTimeFormatter.ISO_OFFSET_DATE_TIME.parse(jobDeployment.getLastDeployedDate());
```

Comment:
Most of those verification seem redundant for the purposes of this test .

You need only to verify 
A) A deploy has finished successfully.  which `    Assertions.assertEquals(testJobVersionSha, jobDeployment.getJobVersion());` would be enough 
B) The deployment has created correctly an image in ECR. I would add a check (if one could be done that is fairly short and easy) that verifies an image exists in ECR.